### ISSUE TITLE
SpeakerDecoder weight normalization

### ISSUE THREAD

**Issue by maevdokimov** [NVIDIA/NeMo#4176]
Hi!

When reading SpeakerDecoder code I bumped into [this](https://github.com/NVIDIA/NeMo/blob/1d3eb3d0d045f7f57316217320b8d265bf3256ce/nemo/collections/asr/modules/conv_asr.py#L848) line. Looks like it is weight normalization for AAS loss. But why should it work?

```
>>> import torch.nn as nn
>>> import torch.nn.functional as F
>>> layer = nn.Linear(128, 100, bias=False)
>>> weight1 = layer.weight.data.clone()
>>> for W in layer.parameters():
...     W = F.normalize(W, p=2, dim=1)
...
>>> weight2 = layer.weight.data.clone()
>>> weight1 == weight2
tensor([[True, True, True,  ..., True, True, True],
        [True, True, True,  ..., True, True, True],
        [True, True, True,  ..., True, True, True],
        ...,
        [True, True, True,  ..., True, True, True],
        [True, True, True,  ..., True, True, True],
        [True, True, True,  ..., True, True, True]])
>>>
```

**Comment by titu1994** [comment]
You should clone your weight before you perform normalize. Right now layer.weights are simply pointing to the same underlying data structure.

**Comment by nithinraok** [comment]
I remember, fiddling with this a long time ago. Though it was not normalizing the weights as I intended, it helped for stable training. Need to investigate more. 

To debug: If you have any dataset, did you try training with and without normalizing final layer weights.

**Comment by maevdokimov** [comment]
But don't you agree that this code does absolutely nothing? The key point of angular loss is to normalize the output projection matrix rows, or else we can get rows with very big norm, that will overbalance other classes and break classification.
